internal/controller/storageclient_controller.go

```diff
@@ -665,6 +710,36 @@ func removeStorageClaimAsOwner(obj client.Object) {
 	}
 }
 
+func (r *storageClientReconcile) reconcileResourceByGVK(
+	gvk schema.GroupVersionKind,
+	obj client.Object,
+	resourceMapping gvkToRawresources,
+) error {
+	resources := resourceMapping[gvk.String()]
+	retainKeys := make(map[types.NamespacedName]bool, len(resources))
+	for idx := range resources {
+		if err := r.reconcileResource(obj, resources[idx]); err != nil {
+			return err
+		}
+		retainKeys[client.ObjectKeyFromObject(obj)] = true
+	}
+	existingObjList := &metav1.PartialObjectMetadataList{}
+	existingObjList.SetGroupVersionKind(gvk)
+	if err := r.list(existingObjList); err != nil {
+		return err
+	}
+	var combinedErr error
+	for idx := range existingObjList.Items {
+		obj := &existingObjList.Items[idx]
+		if metav1.IsControlledBy(obj, &r.storageClient) && !retainKeys[client.ObjectKeyFromObject(obj)] {
+			if err := r.Delete(r.ctx, obj); client.IgnoreNotFound(err) != nil {
+				multierr.AppendInto(&combinedErr, err)
+			}
+		}
+	}
```

Comment:
If we are already running over the entire list and you already have a need for a map, why not just have one loop running on all of the resources and select the action based of the existence of bytes for that object in the server response?